packages/starlight/style/markdown.css

```diff
@@ -62,10 +62,10 @@
 	font-size: var(--sl-text-h6);
 }
 
-.sl-markdown-content a:not(:where(.not-content *)) {
+.sl-markdown-content a:not(:where(:is(.not-content, .not-content *))) {
 	color: var(--sl-color-text-accent);
 }
-.sl-markdown-content a:hover:not(:where(.not-content *)) {
+.sl-markdown-content a:hover:not(:where(:is(.not-content, .not-content *))) {
```

Comment:
These changes are used to avoid `<LinkButton>` getting Markdown link styles applied to it by specifying the `not-content` class on the link itself and not a wrapping element. 

I'm wondering if this is a change that could be applied to some other places to avoid relying on a wrapping element to avoid Markdown styles :thinking:

Reply:
I’d like to avoid this if we can to avoid introducing inconsistencies.

The main reason `.not-content` applies to children not the element with the class is due to spacing. Consider an example like this:

```html
<p>Lorem ipsum...</p>
<div class="not-content">Some non-content...</div>
<p>Dolor sic...</p>
```

If `not-content` removed the margin styles from the `<div>` in this example, it would end up squished next to the first paragraph.

I have sometimes considered whether we should have separate rules for escaping spacing vs other styles, but I’m not sure that’s particularly easy to understand or communicate.

What _would_ be better would be to either give the Markdown styles a lower specificity (e.g. wrap each selector in `:where()` maybe?) or move forward with the `@layer` refactor. That would make it less painful for user components (or our own in this case) to escape the default styles without the extra `not-content` class. (`not-content` would still be needed for circumstances that benefit from a full reset, but there wouldn’t be the temptation to use it because otherwise `!important` is needed to enforce a component style.)

Reply:
Thanks, this makes sense, in this case this was a link color only change but there is no guarantee this would stay the same any way. I think in this case, we should be able to just specify the `color` in the various selectors, e.g. `.sl-link-button.primary`, `.sl-link-button.secondary`, and `.sl-link-button.minimal`.